src/meta-srv/src/election/mysql.rs

```diff
@@ -0,0 +1,1553 @@
+                "Invalid value {}, expect node info || {} || expire time",
+                value, LEASE_SEP
+            ),
+        })?;
+    // Given expire_time is in the format 'YYYY-MM-DD HH24:MI:SS.MS'
+    let expire_time = match Timestamp::from_str(expire_time, None) {
+        Ok(ts) => ts,
+        Err(_) => UnexpectedSnafu {
+            violated: format!("Invalid timestamp: {}", expire_time),
+        }
+        .fail()?,
+    };
+    Ok((value.to_string(), expire_time))
+}
+
+#[derive(Debug, Clone, Default)]
+struct MySqlLeaderKey {
+    name: Vec<u8>,
+    key: Vec<u8>,
+    rev: i64,
+    lease: i64,
+}
+
+impl LeaderKey for MySqlLeaderKey {
+    fn name(&self) -> &[u8] {
+        &self.name
+    }
+
+    fn key(&self) -> &[u8] {
+        &self.key
+    }
+
+    fn revision(&self) -> i64 {
+        self.rev
+    }
+
+    fn lease_id(&self) -> i64 {
+        self.lease
+    }
+}
+
+enum Executor<'a> {
+    Default(MutexGuard<'a, MySqlConnection>),
+    Txn(MySqlTransaction<'a>),
+}
+
+impl Executor<'_> {
+    async fn query(&mut self, query: Query<'_, MySql, MySqlArguments>) -> Result<Vec<MySqlRow>> {
+        match self {
+            Executor::Default(client) => {
+                let res = query
+                    .fetch_all(&mut **client)
+                    .await
+                    .context(MySqlExecutionSnafu)?;
+                Ok(res)
+            }
+            Executor::Txn(txn) => {
+                let res = query
+                    .fetch_all(&mut **txn)
+                    .await
+                    .context(MySqlExecutionSnafu)?;
+                Ok(res)
+            }
+        }
+    }
+
+    async fn execute(&mut self, query: Query<'_, MySql, MySqlArguments>) -> Result<u64> {
+        match self {
+            Executor::Default(client) => {
+                let res = query
+                    .execute(&mut **client)
+                    .await
+                    .context(MySqlExecutionSnafu)?;
+                Ok(res.rows_affected())
+            }
+            Executor::Txn(txn) => {
+                let res = query
+                    .execute(&mut **txn)
+                    .await
+                    .context(MySqlExecutionSnafu)?;
+                Ok(res.rows_affected())
+            }
+        }
+    }
+
+    async fn commit(self) -> Result<()> {
+        match self {
+            Executor::Txn(txn) => {
+                txn.commit().await.context(MySqlExecutionSnafu)?;
+                Ok(())
+            }
+            _ => Ok(()),
+        }
+    }
+}
+
+/// PostgreSql implementation of Election.
+pub struct MySqlElection {
+    leader_value: String,
+    client: Mutex<MySqlConnection>,
+    is_leader: AtomicBool,
+    leader_infancy: AtomicBool,
+    leader_watcher: broadcast::Sender<LeaderChangeMessage>,
+    store_key_prefix: String,
+    candidate_lease_ttl_secs: u64,
+    sql_set: ElectionSqlSet,
+}
+
+impl MySqlElection {
+    pub async fn with_mysql_client(
```

Comment:
Does the mysql sqls and features in use compatible across all mysql versions? If not, or for future compatible consideration, I suggest check mysql version and some of the critical options first. We can target for the most widely used mysql version, 5.7, or the current stable 8.0.

Reply:
I ran sqlness test locally for mysql 8.0, and the sqlness test uses a mysql 5.7, so it should be alright for now. Do we need a compactible check or just leave some comments for future changes?

Reply:
I think a check beforehand is needed, to not cause any runtime surprise to users after metasrv is started. WDYT @fengjiachun

Reply:
That sounds good. I think we can check it before the metasrv starts.